openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py

```diff
@@ -114,12 +160,95 @@ def ingest_records(self, **kwargs) -> None:
+                # continue when the task is being stopped by Airflow.
+
+                # If errors have already been caught during processing, raise them
+                # as well.
+                if error_summary := self.get_ingestion_errors():
+                    raise error_summary from error
+                raise
+
+            except Exception as error:
+                ingestion_error = IngestionError(
+                    error, traceback.format_exc(), query_params
+                )
+
+                if self.skip_ingestion_errors:
+                    # Add this to the errors list but continue processing
+                    self.ingestion_errors.append(ingestion_error)
+                    logger.error(f"Skipping batch due to ingestion error: {error}")
+                    continue
+
+                # Commit whatever records we were able to process, and rethrow the
+                # exception so the taskrun fails.
+                self.commit_records()
+                raise error from ingestion_error
+
+            if self.limit and record_count >= self.limit:
+                logger.info(f"Ingestion limit of {self.limit} has been reached.")
+                should_continue = False
+
+        # Commit whatever records we were able to process
+        self.commit_records()
+
+        # If errors were caught during processing, raise them now
+        if error_summary := self.get_ingestion_errors():
+            raise error_summary
+
+    def get_ingestion_errors(self) -> AirflowException | None:
+        """ "
+        If any errors were skipped during ingestion, returns an AirflowException
+        with a summary of all errors.
+
+        It there are no errors to report, returns None.
+        """
+        if self.ingestion_errors:
+            # Log the affected query_params
+            bad_query_params = ", \n".join(
+                [f"{e.query_params}" for e in self.ingestion_errors]
+            )
+            logger.info(
+                "The following query_params resulted in errors: \n"
+                f"{bad_query_params}"
+            )
+            errors_str = "\n".join(
+                e.repr_with_traceback() for e in self.ingestion_errors
+            )
+            return AirflowException(
+                f"The following errors were encountered during ingestion:\n{errors_str}"
+            )
+        return None
+
+    def get_query_params(
+        self, prev_query_params: Optional[Dict], **kwargs
+    ) -> Optional[Dict]:
+        """
+        Returns the next set of query_params for the next request, handling
+        optional overrides via the dag_run conf.
+        """
+        # If we are getting query_params for the first batch and initial_query_params
+        # have been set, return them.
+        if prev_query_params is None and self.initial_query_params:
+            logger.info(
+                "Using initial_query_params from dag_run conf:"
+                f" {self.initial_query_params}"
```

Comment:
Not sure how important or useful this will be, but it seems like the string that gets spit out here uses single quotes rather than double. When I try to paste that directly into the dagrun conf box, it gets upset about the difference in quotes:

![image](https://user-images.githubusercontent.com/10214785/184707847-f358d396-c81d-4843-b1e1-9b179fba38ed.png)

```
[2022-08-15, 19:46:33 UTC] {provider_data_ingester.py:144} INFO - Begin ingestion for ClevelandDataIngester
[2022-08-15, 19:46:33 UTC] {provider_data_ingester.py:234} INFO - Using initial_query_params from dag_run conf: {'cc': '1', 'has_image': '1', 'limit': 100, 'skip': 200}
```

Reply:
Good call -- I had updated the format of the query_params when they're logged in the error message since I figured folks are likely to copy/paste from there, but this might be copied as well.